### PR TITLE
HDDS-16281. Reduce per-key overhead in Recon reprocess tasks

### PR DESCRIPTION
**Pull request by rich7420**
## What changes were proposed in this pull request?

Recon full reprocess tasks repeatedly resolve worker-local accumulation maps through a shared concurrent map for every key. The OBS namespace summary task also performs RocksDB bucket-table point reads for every key, even when consecutive keys belong to the same bucket.

This change:

* caches OBS bucket metadata per worker and reuses the cached bucket object ID;
* uses ThreadLocal worker accumulation maps while retaining a concurrent registry for final merging and flushing;
* applies the worker-map optimization to namespace summary, container-key mapping, and file-size count reprocess paths;
* adds regression coverage for the OBS bucket lookup count.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-16281

## How was this patch tested?

https://github.com/rich7420/ozone/actions/runs/32956525485

* [10 x 10 flaky-test-check](https://github.com/rich7420/ozone/actions/runs/32956562461): 100/100 passed after retrying one Maven Central HTTP 403 infrastructure failure.
* A local before/after task-level benchmark used 100,000 OBS keys across two buckets, two warmups, and five measured runs, repeated in reverse order. Median reprocess time decreased from 2098/2109 ms on upstream/master to 1598/1548 ms with this patch (24-27% lower), while throughput improved by 31-36%.

Generated-by: Codex (GPT-5)

### CODE REVIEW

File: hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryTaskWithOBS.java [hunk 1]
```diff
@@ -63,6 +72,20 @@ false, getBucketLayout(),
...
+    ReconOMMetadataManager omMetadataManager = spy(getReconOMMetadataManager());
+    Table<String, OmBucketInfo> bucketTable = spy(getReconOMMetadataManager().getBucketTable());
+    doReturn(bucketTable).when(omMetadataManager).getBucketTable();
+    NSSummaryTaskWithOBS task = new NSSummaryTaskWithOBS(getReconNamespaceSummaryManager(), omMetadataManager,
+        OZONE_RECON_NSSUMMARY_FLUSH_TO_DB_MAX_THRESHOLD_DEFAULT, 1, 1, 2000);
+
+    getReconNamespaceSummaryManager().clearNSSummaryTable();
+
+    assertThat(task.reprocessWithOBS(omMetadataManager)).isTrue();
+    verify(bucketTable, times(2)).getSkipCache(anyString());
```
Comment (chihsuan) [inline hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryTaskWithOBS.java]: Would it be possible to make the bucket/key setup local to this test? The hard-coded `times(2)` currently depends on the shared fixture containing exactly two buckets, so an unrelated fixture change could break this test.

**Review by chihsuan — COMMENTED** [review]
Thanks for working on this! @rich7420 the improvement looks promising. 👍 

Would it make sense to add a multi-worker test with a low flush threshold to verify that repeated map replacements do not lose or duplicate data?

Optionally, I think it would be useful to benchmark a high number of buckets to measure peak memory usage, since each worker caches every bucket it encounters until reprocessing finishes.

File: hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryTaskWithOBS.java (hunk 1 above)
Comment (rich7420) [reply hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryTaskWithOBS.java]: Now captures the `getSkipCache` args and asserts `doesNotHaveDuplicates()` (each bucket fetched at most once), so it no longer depends on the bucket count. Also moved into a `@Nested` class.
